### ISSUE TITLE
API for storing UI layout on the backend

### ISSUE THREAD

**Issue by akurzhan** [agblox/ui_proto#2]
Tasks
- [x] Add API endpoint
- [x] Integrate async API integration on front end
- [x] Test integration

Blockers
- [x] Enable CORS for localhost testing

We need API to put/get UI layout in JSON.
Fields: 
+ ID of the profile
+ JSON object

The app state contains the list of dashboards that have been created along with their associated charts/feeds.

Alex and Tom would like for the app state to be shared between all users of the app for demo purposes. Tony should be able to configure a dashboard and Alex should be able to see the same configuration.

The app state can be represented as a single JSON object, so all that is needed is a place to read/update this object.

API endpoint added to support this, "/dashboard".

**Comment by IaroslavR** [comment]
@ericdudley hi my part of task been deployed https://github.com/agblox/research-lab/pull/24/ API docs is here - https://lab.agblox.com/api/ui/

**Comment by ericdudley** [comment]
@IaroslavR Would you be able to enable CORS for my testing locally?

**Comment by IaroslavR** [comment]
> @IaroslavR Would you be able to enable CORS for my testing locally?

@ericdudley CORS already enabled
![Selection_001](https://user-images.githubusercontent.com/9788811/58068732-31f81500-7b9b-11e9-84a8-006834b33bcc.png)

**Comment by IaroslavR** [comment]
@ericdudley If you want to test your app you can use [this](https://github.com/agblox/research-lab/blob/master/api/docker-compose.yml) file as base for you compose file.  
[Example](https://github.com/agblox/research-lab/blob/946741bc0ffedfe7201b853e8dcc9a9b00d4f586/Makefile#L7 ) of multiple compose files usage.  
More info (if you are not familiar with `docker-compose`) can be found [here](https://runnable.com/docker/advanced-docker-compose-configuration) and in [docs](https://docs.docker.com/compose/overview/)

**Comment by IaroslavR** [comment]
@ericdudley We use [alembic](https://alembic.sqlalchemy.org/en/latest/) as database migrations tool. You can run migrations this way(**CAVEAT** database `lab` must be created manually befor you run migrations and `alembic` must be installed):
```bash
cd path/to/research-lab
# change uri to actual for your env
POSTGRES_URI="postgresql://postgres:root@localhost:5432/lab" make run-migration 
```

**Comment by IaroslavR** [comment]
@akurzhan @ericdudley About tests and CI. I'll set up Jenkins in nearest future

**Comment by ericdudley** [comment]
> > @IaroslavR Would you be able to enable CORS for my testing locally?
> 
> @ericdudley CORS already enabled
> ![Selection_001](https://user-images.githubusercontent.com/9788811/58068732-31f81500-7b9b-11e9-84a8-006834b33bcc.png)

I did some research. It seems like the browser is doing a preflight OPTIONS request, but the server is requiring authentication for this request and responding with 401. Could you disable authentication for the OPTIONS method? This seems to be the standard.

**Comment by ericdudley** [comment]
@IaroslavR Not sure if you saw this comment elsewhere.
@IaroslavR The OPTIONS request is now returning a 404.

```
Request URL: https://lab.agblox.com/api/dashboards/1
Request Method: OPTIONS
Status Code: 404 
Remote Address: 134.209.11.202:443
Referrer Policy: no-referrer-when-downgrade
```

This HTML snippet reproduces the issue.
```
<html>
  <head> </head>
  <body>
    <h1 id="test">Loading...</h1>
  </body>
  <script>
    window.onload = function() {
      var username = "admin";
      var password = prompt("Please enter password for admin");
      var headers = new Headers();
      headers.set("Authorization", "Basic " + btoa(username + ":" + password));
      var text = document.getElementById("test");
      fetch("https://lab.agblox.com/api/dashboards/1", {
        method: "GET",
        headers: headers
      })
        .then(response => {
          if (response.status !== 200) {
            text.innerText = "Failed with status code " + response.status + "!";
          } else {
            text.innerText = "Success!";
          }
        })
        .catch(failure => (text.innerText = "Failure!"));
    };
  </script>
</html>
```

**Comment by IaroslavR** [comment]
closed #7 in https://github.com/agblox/research-lab/pull/32 
@ericdudley looks working everywere, btw typo in your snippet here https://github.com/agblox/research-lab/pull/32/commits/4e77fe6f554b1d2ea47542ae3cb1a37631c01648

**Comment by ericdudley** [comment]
@IaroslavR Thanks for fixing this, verified to be working.
